### ISSUE TITLE
Receiving "undefined local variable or method `exception`

### ISSUE THREAD

**Issue by jimmiejackson414** [loopstudio/rails-graphql-api-boilerplate#324]
Hi there, thanks for the wonderful boilerplate!

I was testing out some of the error handling, and am trying to see what happens when I run a mutation to create a new user that already exists in the database. Currently I'm returning the error:

```
undefined local variable or method `exception' for GraphqlSchema:Class
```

The errors appear to be happening in the schema file where the `rescue_from` methods are being defined, specifically the Rescue from RecordInvalid.

Any ideas what could be causing this, or how `exception` should be defined? Thanks!

*Just a note: I'd changed the name of the RailsGraphqlApiBoilerplate to GraphqlSchema instead.

**Comment by mconiglio** [comment]
Hey @jimmiejackson414 , the problem is that both the `RecordInvalid` and the `RecordNotFound` exception handlers had issues. They wasn't causing any issues on our end because the `StandardError` handler was defined first, so it was rescuing all the exceptions instead of letting the other handlers to do so. You can check the fix here if you want: #325

**Comment by jimmiejackson414** [comment]
Awesome thanks for the reply!

Another issue I'm having has to do with the same file, in which this error is returned:
```
Uncaught Error: {
  "errors": [
    {
      "message": "Field '__schema' doesn't exist on type 'Query'",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "query IntrospectionQuery",
        "__schema"
      ],
      "extensions": {
        "code": "undefinedField",
        "typeName": "Query",
        "fieldName": "__schema"
      }
    },
    {
      "message": "Fragment FullType was defined, but not used",
      "locations": [
        {
          "line": 26,
          "column": 1
        }
      ],
      "path": [
        "fragment FullType"
      ],
      "extensions": {
        "code": "useAndDefineFragment",
        "fragmentName": "FullType"
      }
    }
  ]
}
    at Object.next (SchemaFetcher.ts:136)
    at b (Observable.js:135)
    at E (Observable.js:179)
    at e.value (Observable.js:235)
    at bundle.esm.js:76
```

I thought it was coming from the `query_analyzer`s, but using the code from #325 that you provided above and removing those two lines still presents this error. I actually got it working with the following, although it sounds like there might still be an issue with the order of rescues?

```
class GraphqlSchema < GraphQL::Schema
  disable_introspection_entry_points if Rails.env.production?
  use GraphQL::Batch

  query(Types::QueryType)
  mutation(Types::MutationType)

  rescue_from(StandardError) do |exception|
    GraphQL::ExecutionError.new(exception, extensions: { code: :internal_server_error })
  end

  rescue_from(ActiveRecord::RecordInvalid) do |exception|
    errors = exception.record.errors.messages.to_json
    GraphQL::ExecutionError.new(errors, extensions: { code: :bad_request })
  end

  rescue_from(ActiveRecord::RecordNotFound) do |exception|
    GraphQL::ExecutionError.new(exception, extensions: { code: :not_found })
  end
end
```

I also receive a deprecation notice in my server that `GraphQL::Analysis::AST is now the default` and can be removed.

Other than that, I'm really enjoying this boiler and thanks a ton for providing it!
